addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -3010,112 +3010,14 @@ export class OdooEditor extends EventTarget {
-         * </div>
-         * ```
-         *
-         * ```js
-         * _getLastNotEditableAncestorOfNotEditable(document.querySelector(".C")) // return "B"
-         * ```
-         */
-        function _getLastNotEditableAncestorOfNotEditable(node, root) {
-            let currentNode = node;
-            let lastEditable;
-            if (!ancestors(node, root).includes(root)) {
-                return;
-            }
-            while (currentNode && currentNode !== root) {
-                if (currentNode.isContentEditable) {
-                    return lastEditable;
-                } else if (currentNode.isContentEditable === false) {
-                    // By checking that the node is contentEditable === false,
-                    // we ensure at the same time that the currentNode is a
-                    // HTMLElement.
-                    lastEditable = currentNode;
-                }
-                currentNode = currentNode.parentElement;
-            }
-            return lastEditable;
-        }
-
-        const startContainerNotEditable = _getLastNotEditableAncestorOfNotEditable(
-            startContainer,
-            this.editable,
-        );
-        const endContainerNotEditable = _getLastNotEditableAncestorOfNotEditable(
-            endContainer,
-            this.editable,
-        );
-        const bothNotEditable = startContainerNotEditable && endContainerNotEditable;
-
-        if (startContainerNotEditable) {
-            if (startContainerNotEditable.previousSibling) {
-                newRange.setStart(
-                    startContainerNotEditable.previousSibling,
-                    startContainerNotEditable.previousSibling.length,
-                );
-                if (bothNotEditable) {
-                    newRange.setEnd(
-                        startContainerNotEditable.previousSibling,
-                        startContainerNotEditable.previousSibling.length,
-                    );
-                }
-            } else {
-                newRange.setStart(startContainerNotEditable.parentElement, 0);
-                if (bothNotEditable) {
-                    newRange.setEnd(startContainerNotEditable.parentElement, 0);
-                }
-            }
-        }
-        if (!bothNotEditable && endContainerNotEditable) {
-            if (endContainerNotEditable.nextSibling) {
-                newRange.setEnd(endContainerNotEditable.nextSibling, 0);
-            } else {
-                newRange.setEnd(...endPos(endContainerNotEditable.parentElement));
-            }
-        }
-        if (startContainerNotEditable || endContainerNotEditable) {
-            selection.removeAllRanges();
-            selection.addRange(newRange);
-        }
-    }
-
     _onMouseup(ev) {
         this._currentMouseState = ev.type;
 
         this._fixFontAwesomeSelection();
 
-        this._fixSelectionOnContenteditableFalse();
+        if (!ev.target.isContentEditable && this.isFirefox) {
```

Comment:
Why only Firefox ? What happens in Chrome, then ?

Reply:
@dmo-odoo I can reproduce this issue only in firefox.In chrome there is no such issue.In Chrome it is working fine. Because of that i write this fix only for firefox.

Reply:
And what is the issue that you are talking about ?

Reply:
In website, if a snippet is `o_not_editable` then selecting and typing will switch to the nearest editable.

Reply:
But can you still select non editable things for copy/pasting in Firefox with this change ? I think it would remove the range automatically.

Reply:
Ok, So i think for copy paste we need to remove ranges in `onDocumentKeyUp`.

Reply:
@visp-odoo I just tested and it seems that you are right. It's a bug, though, it shouldn't behave that way. We should be able to select something in a non-editable area for copy/paste. I'll create a new task for this issue.

Reply:
@dmo-odoo What about this PR.

Reply:
@dmo-odoo I think with this solution we can select `contentEditable = false` node and on click of it we will remove the ranges or collapse selection.

Reply:
@visp-odoo You said it would not work in Firefox, do you have steps to reproduce the behavior that is wrong on Firefox with the current PR so that I can check it ? Thanks

Reply:
@doco-odoo this is video link of runbot branch in firefox : https://www.awesomescreenshot.com/video/21696738?key=1afe72b0cf64f7cfa02e00c7ff133cdb

Step to reproduct:
- first click on editable area.
- After that click on contentEditable false image.
- Type something
- Now you type the characters then it will be shifted to nearest Editable area.